### PR TITLE
Shop now have images

### PR DESCRIPTION
**Pull request by Crazi12345**
i know im amazing

### CODE REVIEW

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
 import dk.sdu.se.f23.InVasion.common.data.World;
 import dk.sdu.se.f23.InVasion.common.services.EntityProcessingService;
 import dk.sdu.se.f23.InVasion.common.services.PluginService;
 
+import java.util.ArrayList;
+import java.util.Objects;
+
 public class WeaponPlugin implements PluginService {
     private String weaponName = "Steven";
+    private Texture texture = new Texture(Gdx.files.internal("Weapon/src/main/resources/TOWER.png"));
```
Comment (Nolle10) [inline Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java:18]:
```suggestion
    private Texture texture = new Texture(Gdx.files.internal("resources/TOWER.png"));
```

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java [hunk 2]
```diff
@@ -1,22 +1,39 @@
...
 import dk.sdu.se.f23.InVasion.common.data.ProcessAt;
 import dk.sdu.se.f23.InVasion.common.data.World;
 import dk.sdu.se.f23.InVasion.common.services.EntityProcessingService;
 import dk.sdu.se.f23.InVasion.common.services.PluginService;
 
+import java.util.ArrayList;
+import java.util.Objects;
+
 public class WeaponPlugin implements PluginService {
     private String weaponName = "Steven";
```
Comment (Nolle10) [inline Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java:17]: What?
```suggestion
    private String weaponName = "";
```

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
+    private Texture texture = new Texture(Gdx.files.internal("Weapon/src/main/resources/TOWER.png"));
+    private int cost = 200;
 
 
     @Override
     public void onEnable(GameData data, World world) {
-        System.out.println("CREATED PLUGIN");
-        world.addWeapon(weaponName);
+        ArrayList<Object> weaponVariables = new ArrayList<>();
+        weaponVariables.add(weaponName);
```
Comment (Nolle10) [inline Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java:25]: Make the name dynamic, based on what tower you have.

File: Weapon/src/main/resources/TOWER.png
Comment (Nolle10) [inline Weapon/src/main/resources/TOWER.png:1]: I don't think this is the tower, I do believe that the white-blood cells where the desired towers.

File: pom.xml [hunk 4]
```diff
@@ -10,7 +10,9 @@
     <modules>
         <module>GameEngine</module>
         <module>AI</module>
+<!--
```
Comment (Nolle10) [inline pom.xml:13]: Why did you out-comment the bullet module?

**Review by Nolle10 — CHANGES_REQUESTED** [review]
I would love to see these changes before we merge it with the "main" branch.

File: GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java [hunk 5]
```diff
@@ -72,16 +75,26 @@ public void draw() {
         int shopWidth= 200;
         sr.rect(1920-shopWidth,0,200,1080);
         sr.end();
-        sr.begin(ShapeRenderer.ShapeType.Filled);
-        sr.setColor(Color.GRAY);
+
+
+        SpriteBatch spriteBatch = new SpriteBatch();
```
Comment (CamillaFrank) [inline GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java:80]: Vi har 1 spritebatch til hele projektet, brug den i stedet for at lave din egen, brug den istedet plzzz. Du kan få fat i den ved gameData.getSpriteBatch()

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java (hunk 1 above)
Comment (Crazi12345) [reply Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java:18]: See i agree that it looks better. But it does not work for some reason. Enemy and player have also been changed to use the repo root, maven do not like it

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java (hunk 2 above)
Comment (Crazi12345) [reply Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java]: Its just a placeholder name, if you dont like it, i dont kow what to tell you ;)

File: Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java (hunk 3 above)
Comment (Crazi12345) [reply Weapon/src/main/java/dk/sdu/se/f23/InVasion/weapon/WeaponPlugin.java:21]: i do not understand

File: Weapon/src/main/resources/TOWER.png
Comment (Crazi12345) [reply Weapon/src/main/resources/TOWER.png:1]: Where they not the player XD

File: pom.xml (hunk 4 above)
Comment (Crazi12345) [reply pom.xml:13]: my bad, glemte at jeg havde gjort det XD

File: GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java (hunk 5 above)
Comment (Crazi12345) [reply GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java:81]: Changed from SpriteBatch spriteBatch = new SpriteBatch();
to
SpriteBatch spriteBatch = gsm.getGameData().getSpriteBatch();

og nu virker det ikke længere.

kan det være at det er fordi den bliver lavet for sent og nu er null?

Comment (Nolle10) [reply GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java:81]: I do believe so, just make a check that says `if gsm.getGameData().getSpriteBatch(); == null{
gsm.getGameData().setSpriteBatch(new SpriteBatch);}

else{
gsm.getGameData().getSpriteBatch();}`

Comment (Crazi12345) [reply GameEngine/src/main/java/dk/sdu/se/f23/InVasion/gamestates/ShopState.java:81]: I do not know how, but this is both the most beautiful code and the most awful code at the same time XD

**Review by Crazi12345 — COMMENTED** [review]
did some, not all

**Review by CamillaFrank — APPROVED** [review]

**Review by Nolle10 — APPROVED** [review]
Я Джо Байден, и я одобряю этот запрос.

**Pull request merged**